source/php/Helper/File.php

```diff
@@ -9,67 +9,154 @@ class File
+        //If not in cache, look for it, if found cache. 
+        if(file_exists($filePath)) {
+            wp_cache_set($uid, true, __FUNCTION__, $expireFound);
+            return true;
         }
 
-        if (!$namespaceFound) {
-            return '';
+        //Opsie, file not found
+        wp_cache_set($uid, false, __FUNCTION__, $expireNotFound); 
+        return false; 
+    }
+
+     /**
+     * Check if a file exists, cache in redis. 
+     *
+     * @param   string  The file path
+     * @param   integer Time to store positive result
+     * @param   integer Time to store negative result
+     *
+     * @return  bool    If the file exists or not.
+     */
+    public static function fileGetContents($filePath, $args = [], $expire = 86400)
+    {
+        //Allow args to be inputted
+        $use_include_path   = isset($args['use_include_path']) ? $args['use_include_path'] : false;
+        $context            = isset($args['context']) ? $args['context'] : null;
+        $offset             = isset($args['offset']) ? $args['offset'] : 0;
+        $length             = isset($args['length']) ? $args['length'] : null;
+
+        //Unique cache value
+        $uid = "mod_file_get_contents_cache_" . md5($filePath . md5(json_encode($args))); 
+
+        //If in cahce, found
+        $cachedContents = $contents = wp_cache_get($uid, __FUNCTION__); 
+        if($cachedContents !== false) {
+            return $cachedContents;
         }
 
-        return $namespace;
+        //If not in cache, look for it, if found cache. 
+        $contents = file_get_contents(
+            $filePath,
+            $use_include_path,
+            $context,
+            $offset,
+            $length
+        ); 
+
+        //Store in cache
+        wp_cache_set($uid, $contents, __FUNCTION__, $expire); 
+
+        //Return results
+        return $contents;
     }
 
     /**
-     * Creates directory if needed
-     * @param  string $path
-     * @return string
+     * Glob files with caching to improve performance.
+     *
+     * This function uses PHP's glob function to search for files that match the given pattern.
+     * It also utilizes caching to store and retrieve the results, improving performance for
+     * repeated calls to the same pattern.
+     *
+     * @param string $pattern      The file glob pattern to search for files.
+     * @param int    $flags        Flags to pass to the glob function (optional, default is 0).
+     * @param int    $expireFound  Cache expiration time for found results in seconds (optional, default is 0, which means no expiration).
+     * @param int    $expireNotFound Cache expiration time for not found results in seconds (optional, default is 86400 seconds or 24 hours).
+     *
+     * @return array|false An array of matched files or false if no matches were found.
      */
-    public static function maybeCreateDir($path)
+    public static function glob($pattern, $flags = 0, $expireFound = 0, $expireNotFound = 86400)
```

Comment:
To clarify the use of this function in other places, I suggest that we give it a more informative name.
```suggestion
    public static function getCachedGlob($pattern, $flags = 0, $expireFound = 0, $expireNotFound = 86400)
```